### PR TITLE
Wconstab/reland pysymint

### PR DESCRIPTION
**Pull request by wconstab**
rebased https://github.com/pytorch/pytorch/pull/79617/ to see if issues are reproducible.

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: docs/source/conf.py [hunk 1]
```diff
@@ -320,6 +320,7 @@
     "Quantize",
     # torch.utils.backcompat
     "Warning",
+    "SymbolicIntNode"
```
Comment (albanD) [inline docs/source/conf.py:323]: We should NOT add anything to this list.
If this is public, then it should have a proper entry on the website.
If it is not public, it should be marked as such and have a leading `_`.

Comment (wconstab) [reply docs/source/conf.py:323]: cc @Krovatkin  - should we change this? 

This was just a reland PR, the original PR was already stamped.

Comment (Krovatkin) [reply docs/source/conf.py:323]: @albanD it's not public. I'll add an underscore.

Comment (albanD) [reply docs/source/conf.py:323]: Thanks!

**Pull request closed without merging**